lib/stdlib/test/peer_SUITE.erl

```diff
@@ -0,0 +1,459 @@
+
+stop_peer(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    {ok, Peer, _Node} = peer:start_link(#{name => peer:random_name(?FUNCTION_NAME),
+        connection => Conn, args => ["-eval", "timer:sleep(60000)."]}),
+    %% shutdown node
+    peer:stop(Peer).
+
+init_debug() ->
+    [{doc, "Test that debug messages in init work"}].
+
+init_debug(Config) when is_list(Config) ->
+    ct:capture_start(),
+    {ok, Peer, _Node} = peer:start_link(#{name => peer:random_name(?FUNCTION_NAME), shutdown => 1000,
+        connection => standard_io, args => ["-init_debug"]}),
+    ct:sleep(200), %% without this sleep, peer is not fast enough to print
+    peer:stop(Peer),
+    ct:capture_stop(),
+    Texts = lists:append([string:trim(Str, trailing, "\r\n") || Str <- ct:capture_get()]),
+    %% every boot script starts with this
+    Expected = "{progress,preloaded}",
+    Actual = lists:sublist(Texts, 1, length(Expected)),
+    ?assertEqual(Expected, Actual).
+
+io_redirect() ->
+    [{doc, "Tests i/o redirection working for std"}].
+
+io_redirect(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    {ok, Peer, _Node} = peer:start_link(#{connection => Conn}),
+    ct:capture_start(),
+    peer:call(Peer, io, format, [user, "test.", []]),
+    peer:call(Peer, erlang, apply, [io, format, ["second."]]),
+    %% ensure no dist connection is set up
+    ?assertNot(lists:member(Peer, nodes()), {dist_connected, Peer, nodes()}),
+    ct:capture_stop(),
+    Texts = ct:capture_get(),
+    peer:stop(Peer),
+    ?assertEqual(["test.", "second."], Texts).
+
+multi_node() ->
+    [{doc, "Tests several nodes starting concurrently"}].
+
+multi_node(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    Peers = [
+        peer:start_link(#{name => peer:random_name(?FUNCTION_NAME),
+            wait_boot => {self(), tag}, connection => Conn})
+        || _ <- lists:seq(1, 4)],
+    Nodes = [receive {tag, {started, Node, Peer}} -> Node end || {ok, Peer} <- Peers],
+    ?assertEqual(4, length(Nodes)),
+    [?assertEqual(ok, peer:stop(Peer)) || {ok, Peer} <- Peers].
+
+duplicate_name() ->
+    [{doc, "Tests that a node with the same name fails to start"}].
+
+duplicate_name(Config) when is_list(Config) ->
+    Conn = proplists:get_value(connection, Config),
+    {ok, Peer, _Node} = peer:start_link(#{connection => Conn, name => ?FUNCTION_NAME, register => false}),
+    ?assertException(exit, _, peer:start_link(#{connection => standard_io, name => ?FUNCTION_NAME})),
+    peer:stop(Peer).
+
+%% -------------------------------------------------------------------
+%% Compatibility: old releases
+old_release() ->
+    [{doc, "Verity running with previous OTP release"}].
+
+old_release(Config) when is_list(Config) ->
+    PrivDir = proplists:get_value(priv_dir, Config),
+    ?assertEqual(not_available, ?CT_PEER([], "1", PrivDir)),
```

Comment:
A bug in `test_server_node` results in the pattern `"1"` finding an OTP 19 release (which will not work because it does not have `is_map_key/2`). I don't have to go down the rabbit hole of fixing bus in `test_server_node`, so here is a workaround that avoids the bug: 

```suggestion
    ?assertEqual(not_available, ?CT_PEER([], "10", PrivDir)),
```